### PR TITLE
Wrote tests for browse.go and redir.go

### PR DESCRIPTION
**Pull request by PatelNDipen**

### CODE REVIEW

File: caddy/setup/browse_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		"browse /",
+		"browse . " + tempTemplatePath,
+		"browse . " + nonExistantDirPath,
+		"browse " + tempDirPath + "\n browse " + tempDirPath,
+	}
+
+	tests := []struct {
+		expectedPathScope []string
+		shouldErr         bool
+	}{
```
Comment (mholt) [inline caddy/setup/browse_test.go]: How about making the input another field in this struct? In other words, combine `testTokens` and `tests`.

Then you can actually iterate an anonymous slice below instead, like this:

```go
for i, test := range []struct{
    // input, expectedPathScope, shouldErr fields here
}{
    // ... test cases here ...
}{
    // test logic here
}
```

Can use a similar strategy for redir_test.go too.

**Comment by PatelNDipen** [comment]
Ok thanks for the tips. I'll make those adjustments soon.

File: caddy/setup/browse_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		{"browse . " + tempTemplatePath, []string{"."}, false},
+
+		// test case #3 tests detection of non-existant template
+		{"browse . " + nonExistantDirPath, nil, true},
+
+		// test case #4 tests detection of duplicate pathscopes
+		{"browse " + tempDirPath + "\n browse " + tempDirPath, nil, true},
+	} {
+
+		//		c := &Controller{Config: &server.Config{Root: "."}, Dispenser: parse.NewDispenser("", strings.NewReader(testTokens[i]))}
```
Comment (mholt) [inline caddy/setup/browse_test.go]: Not to nit-pick :smile: but this commented line can probably go away

Comment (PatelNDipen) [inline caddy/setup/browse_test.go]: Sorry, I should have removed that before committing.

**Comment by mholt** [comment]
Excellent, these look great. Thanks, Patel!

**Pull request merged**